src/selectors/expandThoughts.ts

```diff
@@ -50,23 +50,41 @@ function expandThoughts<B extends boolean>(state: State, path: Path | null, opti
  *   ...
  * }
  */
-function expandThoughts(state: State, path: Path | null, { depth = 0, returnContexts }: { depth?: number, returnContexts?: boolean } = {}): Index<Path | Context> {
+function expandThoughts (state: State, path: Path | null, options?: { suppressExpansion?: boolean, returnContexts?: boolean }): Index<Path | Context> {
+
+  /** Get the ancestor of the given path from which the expansion should start. */
+  const startingPath = (() => {
+    const ancestorPath = path && path.slice(0, -MAX_DISTANCE_FROM_CURSOR)
+    return ancestorPath && ancestorPath.length !== 0 ? ancestorPath : HOME_PATH
+  })()
+
+  return expandThoughtsRecursive(state, path || HOME_PATH, startingPath, options)
+}
+
+/**
+ * Recursively generate expansion map based by checking if the children of the current path should expand based on the given expansion path.
+ *
+ * @param expansionPath - The path based on which expansion happens.
```

Comment:
Let's call this `startingPath` or `expansionBasePath` to be a bit more specific.

Reply:
Sure.